### PR TITLE
[DataGridPro] Fix missing `styleOverrides` on pinned columns

### LINKED ISSUE #3716: Class pinnedColumns can't be overwritten
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

I try to style the pinnedColumns class with a customer style via createTheme and ThemeProvider.
Other styles can be overwritten but the pinnedColumns class doesn't seem to be accessible.

```javascript
 MuiDataGrid: {
          styleOverrides: {
            // This works
            root: {
              backgroundColor: '#fff',
            },
            // This works too
            pinnedColumnHeaders: {
              backgroundColor: '#fff',
              boxShadow: 'none',
            },
            // This doesn't work
            pinnedColumns: {
              backgroundColor: '#fff',
              boxShadow: 'none',
            },
          },
        },
```

### Expected behavior 🤔

We should be able to theme pinnedColumns too.

### Order ID 💳 (optional)

35878

### PR DESCRIPTION
**Pull request by alexfauquette**
Fix #3716 

CodeSandbox with the fix: https://codesandbox.io/s/disablecolumnpinningbuttons-material-demo-forked-se9cx?file=/demo.tsx

### CODE REVIEW

File: packages/grid/_modules_/grid/gridClasses.ts [hunk 1]
```diff
@@ -389,6 +389,9 @@ export const gridClasses = generateUtilityClasses('MuiDataGrid', [
   'pinnedColumns',
   'pinnedColumns--left',
   'pinnedColumns--right',
+  'pinnedColumnHeaders',
```
Comment (m4theushw) [inline packages/grid/_modules_/grid/gridClasses.ts:392]: You also need to add these classes to the interface in this same file. There're should be docs changes in this PR.

Comment (alexfauquette) [reply packages/grid/_modules_/grid/gridClasses.ts:404]: Is there a reason why we don't reuse the interface keys to generate the list of classes.

I found 3 elements in `gridClasses` which are not parts of the interface, and I don't know if it is expected or not :
- 'actionsCell'
- 'groupingCriteriaCell'
- 'groupingCriteriaCellToggle'

Comment (m4theushw) [reply packages/grid/_modules_/grid/gridClasses.ts:404]: I copied the same approach from MUI Core. I don't know how they check to avoid missing classes.

**Review by m4theushw — APPROVED** [review]
It's working.

**Pull request merged**